lib/Stor.pm

```diff
@@ -212,18 +208,21 @@ sub _sha_to_filepath($self, $sha) {
 
 
 sub _stream_found_file($self, $c, $path) {
-
+    my $tm_start = time;
     my $fh = $path->openr_raw();
     my $drain; $drain = sub {
         my ($c) = @_;
 
         my $chunk;
-        if (read($fh, $chunk, 1024 * 1024) == 0) {
+        my $size = read($fh, $chunk, 1024 * 1024);
+        if ($size == 0) {
             close($fh);
             $drain = undef;
         }
 
-        $c->write($chunk, $drain)
+        $c->write($chunk, $drain);
+        $self->statsite->update('success.get.ok.size', $size);
+        $self->statsite->timing('success.get.ok.time', (time - $tm_start) / 1000);
```

Comment:
Shouldn't be time logged only when `$size < 1024*1024`? Otherwise you sent as many metrics as chunk you send to client.

Reply:
it isn't possible... this time metrics means how long is `$size` sending...

Reply:
The method `_stream_found_file` or `drain` is called for each 1MB chunk of the file? If `_stream_found_file` than it's time for sending this chunk (_OK_) but if `drain` than you measure time from starting downloading of the file to the end of each chunk (_INCORRECT_).

Reply:
I don't know =) I will try in minimal test...

Reply:
I try it and fix it =)